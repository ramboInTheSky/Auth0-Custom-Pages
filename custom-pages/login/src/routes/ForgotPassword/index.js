import ForgotPassword from './components/ForgotPasswordForm'

// Sync route definition
export default {
  path: 'forgotPassword',
  component: ForgotPassword
}
