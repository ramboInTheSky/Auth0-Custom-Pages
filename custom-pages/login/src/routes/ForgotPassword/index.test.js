import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-14'
import ForgotPassword from './'

Enzyme.configure({ adapter: new Adapter() })

describe('ForgotPassword components', () => {
  let el
  let instance
  const auth = {
    forgotPassword: jest.fn()
  }
  const functionSignature = 'a function signature'
  jest.mock('../../services/AuthApi')

  beforeEach(() => {
    el = shallow(<ForgotPassword.component />)
    instance = el.instance()
    instance.auth = auth
    instance.requestErrorCallback = functionSignature
  })

  it('should render <ForgotPassword /> correctly', () => {
    expect(el).toMatchSnapshot()
  })

  it('should instantiate AuthApi', () => {
    expect(instance.auth).toBe(auth)
  })

  it('should render correctly when there is an error', () => {
    el.setState({ error: 'this is an error' }).update()
    expect(el).toMatchSnapshot()
  })

  it('should render correctly when there is email', () => {
    el.setState({
      valid: true
    }).update()
    expect(el).toMatchSnapshot()
  })

  it('should call the AuthApi forgotPAssword when login is invoked', () => {
    const email = 'username@email.com'
    el.setState({
      email
    }).update()

    instance.forgotPassword()
    expect(instance.auth.forgotPassword).toBeCalledWith(email, functionSignature)
  })
})
