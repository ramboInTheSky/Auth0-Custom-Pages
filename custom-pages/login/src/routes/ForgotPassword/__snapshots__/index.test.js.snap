// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword components should render <ForgotPassword /> correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Forgot password
        </h3>,
        <h5>
          If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
        </h5>,
        <form
          className=""
        >
          <Fieldset
            legend="Personal information"
          >
            <Input
              data-qa-sel="forgotPassword-email"
              id="email"
              label="Email"
              name="email"
              onChange={[Function]}
              placeholder="eg: your.name@example.com..."
              type="email"
            />
          </Fieldset>
          <button
            className="btn btn--cta"
            data-qa-sel="forgotPassword-button"
            disabled={true}
            onClick={[Function]}
          >
            Reset password
          </button>
        </form>,
        <Link
          data-qa-sel="forgotPassword-link-to-login"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/"
        >
          Return to sign in
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Forgot password",
        },
        "ref": null,
        "rendered": "Forgot password",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        },
        "ref": null,
        "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Fieldset
              legend="Personal information"
            >
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>,
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                null,
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />,
              ],
              "legend": "Personal information",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data-qa-sel": "forgotPassword-email",
                  "id": "email",
                  "label": "Email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "eg: your.name@example.com...",
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset password",
              "className": "btn btn--cta",
              "data-qa-sel": "forgotPassword-button",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset password",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Return to sign in",
          "data-qa-sel": "forgotPassword-link-to-login",
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "/",
        },
        "ref": null,
        "rendered": "Return to sign in",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Forgot password
          </h3>,
          <h5>
            If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
          </h5>,
          <form
            className=""
          >
            <Fieldset
              legend="Personal information"
            >
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>
          </form>,
          <Link
            data-qa-sel="forgotPassword-link-to-login"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Return to sign in
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Forgot password",
          },
          "ref": null,
          "rendered": "Forgot password",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          },
          "ref": null,
          "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Fieldset
                legend="Personal information"
              >
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />
              </Fieldset>,
              <button
                className="btn btn--cta"
                data-qa-sel="forgotPassword-button"
                disabled={true}
                onClick={[Function]}
              >
                Reset password
              </button>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  null,
                  <Input
                    data-qa-sel="forgotPassword-email"
                    id="email"
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="eg: your.name@example.com..."
                    type="email"
                  />,
                ],
                "legend": "Personal information",
              },
              "ref": null,
              "rendered": Array [
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "data-qa-sel": "forgotPassword-email",
                    "id": "email",
                    "label": "Email",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "eg: your.name@example.com...",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset password",
                "className": "btn btn--cta",
                "data-qa-sel": "forgotPassword-button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset password",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Return to sign in",
            "data-qa-sel": "forgotPassword-link-to-login",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/",
          },
          "ref": null,
          "rendered": "Return to sign in",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`ForgotPassword components should render correctly when there is an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Forgot password
        </h3>,
        <h5>
          If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
        </h5>,
        <form
          className=""
        >
          <Fieldset
            legend="Personal information"
          >
            <Alert
              type="error"
            >
              this is an error
            </Alert>
            <Input
              data-qa-sel="forgotPassword-email"
              id="email"
              label="Email"
              name="email"
              onChange={[Function]}
              placeholder="eg: your.name@example.com..."
              type="email"
            />
          </Fieldset>
          <button
            className="btn btn--cta"
            data-qa-sel="forgotPassword-button"
            disabled={true}
            onClick={[Function]}
          >
            Reset password
          </button>
        </form>,
        <Link
          data-qa-sel="forgotPassword-link-to-login"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/"
        >
          Return to sign in
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Forgot password",
        },
        "ref": null,
        "rendered": "Forgot password",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        },
        "ref": null,
        "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Fieldset
              legend="Personal information"
            >
              <Alert
                type="error"
              >
                this is an error
              </Alert>
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>,
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Alert
                  type="error"
                >
                  this is an error
                </Alert>,
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />,
              ],
              "legend": "Personal information",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "this is an error",
                  "type": "error",
                },
                "ref": null,
                "rendered": "this is an error",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data-qa-sel": "forgotPassword-email",
                  "id": "email",
                  "label": "Email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "eg: your.name@example.com...",
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset password",
              "className": "btn btn--cta",
              "data-qa-sel": "forgotPassword-button",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset password",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Return to sign in",
          "data-qa-sel": "forgotPassword-link-to-login",
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "/",
        },
        "ref": null,
        "rendered": "Return to sign in",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Forgot password
          </h3>,
          <h5>
            If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
          </h5>,
          <form
            className=""
          >
            <Fieldset
              legend="Personal information"
            >
              <Alert
                type="error"
              >
                this is an error
              </Alert>
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>
          </form>,
          <Link
            data-qa-sel="forgotPassword-link-to-login"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Return to sign in
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Forgot password",
          },
          "ref": null,
          "rendered": "Forgot password",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          },
          "ref": null,
          "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Fieldset
                legend="Personal information"
              >
                <Alert
                  type="error"
                >
                  this is an error
                </Alert>
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />
              </Fieldset>,
              <button
                className="btn btn--cta"
                data-qa-sel="forgotPassword-button"
                disabled={true}
                onClick={[Function]}
              >
                Reset password
              </button>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Alert
                    type="error"
                  >
                    this is an error
                  </Alert>,
                  <Input
                    data-qa-sel="forgotPassword-email"
                    id="email"
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="eg: your.name@example.com..."
                    type="email"
                  />,
                ],
                "legend": "Personal information",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "this is an error",
                    "type": "error",
                  },
                  "ref": null,
                  "rendered": "this is an error",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "data-qa-sel": "forgotPassword-email",
                    "id": "email",
                    "label": "Email",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "eg: your.name@example.com...",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset password",
                "className": "btn btn--cta",
                "data-qa-sel": "forgotPassword-button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset password",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Return to sign in",
            "data-qa-sel": "forgotPassword-link-to-login",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/",
          },
          "ref": null,
          "rendered": "Return to sign in",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`ForgotPassword components should render correctly when there is email 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Forgot password
        </h3>,
        <h5>
          If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
        </h5>,
        <form
          className=""
        >
          <Fieldset
            legend="Personal information"
          >
            <Input
              data-qa-sel="forgotPassword-email"
              id="email"
              label="Email"
              name="email"
              onChange={[Function]}
              placeholder="eg: your.name@example.com..."
              type="email"
            />
          </Fieldset>
          <button
            className="btn btn--cta"
            data-qa-sel="forgotPassword-button"
            disabled={true}
            onClick={[Function]}
          >
            Reset password
          </button>
        </form>,
        <Link
          data-qa-sel="forgotPassword-link-to-login"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/"
        >
          Return to sign in
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Forgot password",
        },
        "ref": null,
        "rendered": "Forgot password",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        },
        "ref": null,
        "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Fieldset
              legend="Personal information"
            >
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>,
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                null,
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />,
              ],
              "legend": "Personal information",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data-qa-sel": "forgotPassword-email",
                  "id": "email",
                  "label": "Email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "eg: your.name@example.com...",
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset password",
              "className": "btn btn--cta",
              "data-qa-sel": "forgotPassword-button",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset password",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Return to sign in",
          "data-qa-sel": "forgotPassword-link-to-login",
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "/",
        },
        "ref": null,
        "rendered": "Return to sign in",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Forgot password
          </h3>,
          <h5>
            If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.
          </h5>,
          <form
            className=""
          >
            <Fieldset
              legend="Personal information"
            >
              <Input
                data-qa-sel="forgotPassword-email"
                id="email"
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="eg: your.name@example.com..."
                type="email"
              />
            </Fieldset>
            <button
              className="btn btn--cta"
              data-qa-sel="forgotPassword-button"
              disabled={true}
              onClick={[Function]}
            >
              Reset password
            </button>
          </form>,
          <Link
            data-qa-sel="forgotPassword-link-to-login"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Return to sign in
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Forgot password",
          },
          "ref": null,
          "rendered": "Forgot password",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          },
          "ref": null,
          "rendered": "If you have forgotten your password, please use the password reset feature below. Enter the email address you registered with below and click the reset password button. If your account can be found we will send you a email with a link you can use to reset your password.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Fieldset
                legend="Personal information"
              >
                <Input
                  data-qa-sel="forgotPassword-email"
                  id="email"
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="eg: your.name@example.com..."
                  type="email"
                />
              </Fieldset>,
              <button
                className="btn btn--cta"
                data-qa-sel="forgotPassword-button"
                disabled={true}
                onClick={[Function]}
              >
                Reset password
              </button>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  null,
                  <Input
                    data-qa-sel="forgotPassword-email"
                    id="email"
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    placeholder="eg: your.name@example.com..."
                    type="email"
                  />,
                ],
                "legend": "Personal information",
              },
              "ref": null,
              "rendered": Array [
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "data-qa-sel": "forgotPassword-email",
                    "id": "email",
                    "label": "Email",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "eg: your.name@example.com...",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset password",
                "className": "btn btn--cta",
                "data-qa-sel": "forgotPassword-button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset password",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Return to sign in",
            "data-qa-sel": "forgotPassword-link-to-login",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/",
          },
          "ref": null,
          "rendered": "Return to sign in",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
